crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -465,72 +526,247 @@ where
         &mut self,
         process: &ProcessState,
     ) -> impl FutureMaybeSend<Result<Vec<AdviceMutation>, EventError>> {
-        let event_id = EventId::from_felt(process.get_stack_item(0));
+        // TODO: We can technically avoid putting the &ProcessState reference here into the future,
+        // but it seems to work and maybe we should try building the web client against this branch
+        // to test if we can do this, as it does result in more readable code.
+        async move {
```

Comment:
I would try to move it both the handling of stdlib events and `TransactionEvent` construction out of the async context. It will be a bit more code - but I think it is conceptually cleaner this way:

1. First we handle stdlib events.
2. Then we read w/e data we need from the process/host.
3. And only then we enter async context to make any required async requests.

Reply:
I moved it outside so you can see what it looks like, but because we cannot return anything outside the async block, this is not as pretty overall.

Reply:
Yes, makes sense - but I'd still probably go with this approach (the additional code complexity is pretty small IMO).